subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/KotlinSettingsScript.kt

```diff
@@ -73,7 +74,8 @@ import kotlin.script.templates.ScriptTemplateDefinition
         "-Xjsr305=strict",
         "-XXLanguage:+DisableCompatibilityModeForNewInference",
         "-XXLanguage:-TypeEnhancementImprovementsInStrictMode",
-    ]
+    ],
+    provider = KotlinBuildScriptTemplateAdditionalCompilerArgumentsProvider::class
```

Comment:
❔ I guess there's no way to extract these annotations to a super-class, right?

Reply:
Forgot to answer this one: I tried it, but it seems it doesn't work with a super-class.